### PR TITLE
paste geojson features

### PR DESCRIPTION
**Pull request by missinglink**
The GeoJSON Region Coverer is now much more tolerant of invalid/funky geometries, we might want to add an option to paste geojson:

Screenshot 2024-09-23 at 00 38 25

Screenshot 2024-09-23 at 00 35 33

### CODE REVIEW

**Comment by missinglink** [comment]
There is a minor issue with this currently, if you paste geojson and then hit the up/down arrows on min/max cell it doesn't refresh like it does with drawing, unless you reload the json text area.

File: src/App.tsx [hunk 1]
```diff
@@ -224,9 +224,12 @@ function App() {
   const [maxCells, setMaxCells] = createSignal(200);
   const [drawMode, setDrawMode] = createSignal("");
   const [cellUnionText, setCellUnionText] = createSignal("");
```
Comment (bdon) [inline src/App.tsx:226]: can we get away with a single textbox that detects if the input starts with `{` for GeoJSON?

**Comment by bdon** [comment]
I think the GeoJSON needs to be loaded into terra-draw state so it is persistent, instead of just being parsed once when it is loaded from the text box.

**Comment by missinglink** [comment]
> needs to be loaded into terra-draw state so it is persistent

hmm I just tried this and it's kinda tricky because terradraw attempts to validate the geometry:

Screenshot 2024-09-23 at 11 06 38


```js
const feature = JSON.parse(jsonArea.value);
if (feature?.type !== "Feature") throw new Error("Invalid Feature");

feature.id = "ffaae8e0-cf4b-47f9-9cf2-4bcaa9826a2b";
feature.properties = { mode: "polygon" };

draw!.clear();
draw!.addFeatures([feature]);
computeCoveringForDraw();
```

It only seems to support a subset of the GeoJSON geometry types:

Screenshot 2024-09-23 at 11 13 43

**Comment by bdon** [comment]
Can you normalize a MultiPolygon into a list of Polygon features, etc?

**Comment by missinglink** [comment]
> Can you normalize a MultiPolygon into a list of Polygon features, etc

I tried this and it's difficult because terradraw still tries to validate the features according to the drawing mode, I couldn't get it to work.

So what I ended up doing is storing the features in another variable outside terradraw.

It works now, but feels kinda messy 🤷‍♂️

**Pull request merged**